app/services/vacancy_facets.rb

```diff
@@ -17,6 +17,10 @@ def counties
     cached(:counties) { sort_and_limit(county_facet, 20) }
   end
 
+  def additional_job_roles
+    cached(:job_roles) { job_role_facet.select { |role| role.in?(Vacancy.additional_job_role_options) }.to_h }
```

Comment:
You will need to give a different key to the `cached` method, otherwise it'll just return cached data from the job roles - e.g. `cached(:additional_job_roles) { ... }`